### PR TITLE
Remove ResourceType from DeploymenResources

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

This touches many parts of the project, in order to remove the deprecated ResourceType from the DeploymentResources.
It had to be removed from the Protocol interfaces but also from the Gateway protocol, from go client etc.

In order to achieve this I needed to disable some backwards compatible checks. I assume we want to bring them back after 1.0, should I create follow up issues for it? Do we want to bring it also back for the go client? (no public api?)

[Note: Operate is already informed about this change. ](https://camunda.slack.com/archives/C9B5270DA/p1619438194188800)

## Related issues

related #6863 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by Zelldon** [comment]
@npepinpe would be cool if we could have this in the alpha

**Comment by npepinpe** [comment]
Unfortunately I think @MiguelPires is already building the alpha? Could be wrong

File: .ci/scripts/distribution/build-go.sh [hunk 1]
```diff
@@ -17,7 +17,8 @@ for file in {internal,cmd/zbctl/internal}/*; do
   EXCLUDE="$EXCLUDE --exclude-package $PREFIX/$file"
 done
 
-/usr/bin/gocompat compare --go1compat $EXCLUDE ./...
+# Disabled for 1.0
+# /usr/bin/gocompat compare --go1compat $EXCLUDE ./...
```
Comment (npepinpe) [inline .ci/scripts/distribution/build-go.sh:21]: There's a follow up issue for clirr, can you make a comment there as well to reenable this?

File: clients/go/README.md [hunk 2]
```diff
@@ -11,6 +16,12 @@ To regenerate the gateway mock `internal/mock_pb/mock_gateway.go` run [`mockgen`
 GO111MODULE=off mockgen github.com/camunda-cloud/zeebe/clients/go/pkg/pb GatewayClient,Gateway_ActivateJobsClient > internal/mock_pb/mock_gateway.go
 ```
 
+If you see errors regarding packages which are not found then as alternative you can try this:
+
+```
+GO111MODULE=off mockgen -source $GOPATH/src/github.com/camunda-cloud/zeebe/clients/go/pkg/pb/gateway.pb.go GatewayClient,Gateway_ActivateJobsClient > internal/mock_pb/mock_gateway.go
```
Comment (npepinpe) [inline clients/go/README.md:22]: :+1:

File: parent/pom.xml [hunk 3]
```diff
@@ -1097,6 +1097,7 @@
           <version>${plugin.version.proto-backwards-compatibility}</version>
           <configuration>
             <protoSourceRoot>${project.basedir}/src/main/proto</protoSourceRoot>
+            <allowBreakingChanges>true</allowBreakingChanges>
```
Comment (npepinpe) [inline parent/pom.xml:1100]: same - can we add a comment in the clirr issue to remove this later (or make a new follow up issue)?

**Review by npepinpe — APPROVED** [review]
Two small things, not blockers: can we either have a follow up issue to reenable backwards compat checks, or add a comment in the existing clirr issue (sort of relevant) to also do this? Thanks!

**Comment by MiguelPires** [comment]
> Unfortunately I think @MiguelPires is already building the alpha? Could be wrong

I was but I can wait for this PR if you want

**Comment by Zelldon** [comment]
@MiguelPires this would be awesome

**Comment by Zelldon** [comment]
Bors r+

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**